CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+cmake_minimum_required(VERSION 3.16)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+option(SANITIZE_ADDRESS "Address Sanitizer." OFF)
+
+if (SANITIZE_ADDRESS)
+set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+endif()
+
+include(FetchContent)
+
+if(NOT InstallVendor)
+function(InstallVendor project_name git_repo branch)
+        IF (NOT ${project_name})
+                FetchContent_Declare(
+                        ${project_name}
+                        GIT_REPOSITORY ${git_repo}
+                        GIT_TAG ${branch}
+                        SOURCE_DIR "${PROJECT_SOURCE_DIR}/vendor/${project_name}"
+                        BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/${project_name}"
+                        SUBBUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/${project_name}-sub"
+                )
+                FetchContent_MakeAvailable(${project_name})
+                set(${project_name} CACHE INTERNAL "")
+                message(STATUS "Found ${project_name} @ ${branch}")
+        endif()
+endfunction()
+endif()
+
+set(This oof)
+project(${This} LANGUAGES C CXX)
+
+#InstallVendor(doctest https://github.com/onqtam/doctest.git master)
+
+set(LIBRARY_HEADERS
+${PROJECT_SOURCE_DIR}/oof.h
+)
+
+file(GLOB_RECURSE DEMO_HEADERS ${PROJECT_SOURCE_DIR}/demos/**.h)
+file(GLOB_RECURSE DEMO_SOURCES ${PROJECT_SOURCE_DIR}/demos/**.cpp)
+
+file(GLOB_RECURSE TEST_SOURCES ${PROJECT_SOURCE_DIR}/tests/**.cpp)
+
+add_library(${This} INTERFACE)
```

Comment:
To allow vendoring this project, the target name should have a project specific prefix which is stripped in the install interface. You can find such an example [here](https://github.com/friendlyanon/cmake-init-header-only/blob/master/CMakeLists.txt#L20-L26).

Reply:
I thought this was quite a common way of doing it? But you might be right in that there is no reason for namespaces in that case. Will do.

Reply:
If you check Craig Scott's Deep CMake talk slides, he suggests the same at the end. He didn't have time to mention it in the talk though.